contracts/access/AccessController.sol

```diff
@@ -141,8 +123,8 @@ contract AccessController is IAccessController, AccessManagedUpgradeable, UUPSUp
         if (permission > 2) {
             revert Errors.AccessController__PermissionIsNotValid();
         }
-        if (!IModuleRegistry($.moduleRegistry).isRegistered(msg.sender) && ipAccount != msg.sender) {
-            revert Errors.AccessController__CallerIsNotIPAccount();
+        if (ipAccount != msg.sender && IIPAccount(payable(ipAccount)).owner() != msg.sender) {
```

Comment:
I think we could remove the 1st parameter of `setPermission` (`ipAccount`) in favor of msg.sender, and here you check that sender is an ipAccount or the owner

Reply:
iiuc, Do you mean remove the `ipAccount` from the `setPermission()` parameters? 
Line 112-114 is for checking whether `ipAccount` is address(0).
As `ipAccount` is a field of permission, we need to store it in the permission mapping; hence, I don't feel we could remove it.

Reply:
Yeah, reviewing it, the thing is that this method can be called by the ipAccount or the owner. I read it as only callable by IPAccount. Nevermind!